packages/terra-button/src/Button.jsx

```diff
@@ -174,7 +185,7 @@ class Button extends React.Component {
 
     // Add focus styles for keyboard navigation
     if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
-      this.setState({ focused: true });
+      this.setState(prevState => ({ focused: true, isSelected: !prevState.isSelected }));
```

Comment:
I would move this out to different block and update only if isSelectable is true to prevent unintended re-render of component

Reply:
Added check in https://github.com/cerner/terra-core/pull/3940/commits/54c4c362b16f27f2bdfc0acb91705ac1e11d9c7e#